test/samplers/test_samplers.py

```diff
@@ -131,6 +134,80 @@ def test_sampling_range_negative(sampler):
         assert_tensor_equal(clip.data, clips_1[0].data)
 
 
+@pytest.mark.parametrize("sampler", (clips_at_random_indices, clips_at_regular_indices))
+def test_sampling_range_default_behavior(sampler):
+    # This is a functional test for the default behavior of the
+    # sampling_range_end parameter. By default it's None, which means the
+    # sampler automatically sets its value such that we never sample "beyond"
+    # the number of frames in the video. That means that the last few frames of
+    # the video are less likely to be part of a clip.
+    # When sampling_range_end is set manually to e.g. len(decoder), the last
+    # frames are way more likely to be part of a clip, since there is no
+    # restriction on the sampling range (and the user-defined policy comes into
+    # action, potentially repeating that last frame).
+    #
+    # In this test we assert that the last sampled frame occurs significantly
+    # more often when sampling_range_end=len(decoder) than when it's None.
+    # This is only a proxy, for lack of better testing oppportunities.
```

Comment:
Looking into this more I think I went overboard here. It would be far simpler to just check the *beginning* of the last clip, instead of the last frame across all clips. That's a more direct check of the expected behavior. I'll re-write this test but please don't let that block the review :)